m3u8/model.py

```diff
@@ -1119,6 +1137,57 @@ def dumps(self):
     def __str__(self):
         return self.dumps()
 
+class DateRangeList(TagList):
+    pass
+
+class DateRange(object):
+    def __init__(self, **kwargs):
+        self.id = kwargs.get('id')
+        self.start_date = kwargs.get('start_date')
+        self.class_ = kwargs.get('class')
+        self.end_date = kwargs.get('end_date')
+        self.duration = kwargs.get('duration')
+        self.planned_duration = kwargs.get('planned_duration')
+        self.scte35_cmd = kwargs.get('scte35_cmd')
+        self.scte35_out = kwargs.get('scte35_out')
+        self.scte35_in = kwargs.get('scte35_in')
+        self.end_on_next = kwargs.get('end_on_next')
+        self.x_client_attrs = [ (attr, kwargs.get(attr)) for attr in kwargs if attr.startswith('x_') ]
+
+    def dumps(self):
+        daterange = []
+        daterange.append('ID=' + quoted(self.id))
+
+        if (self.start_date):
```

Comment:
This is required, right? Could not it be written just like the `ID` field above?

Reply:
I thought I had put a comment here to explain why this is so, but I must have forgotten.

The spec says it's required, but there's also an example in the spec that does not contain this (see https://tools.ietf.org/html/rfc8216#section-8.10). Further, we see examples in the wild where this attribute is not used, presumably because of the example mentioned above, so making it required is problematic.

I'll add a comment.

Reply:
Note this case is covered in the provided tests (by using a derivation of the example in the RFC).

Reply:
Comment added in 22bce1e.

Reply:
I agree with the comment. Thank you for clarifying this